src/mod/gcewing/sg/tileentity/SGPowerTE.java

```diff
@@ -78,43 +59,50 @@ public boolean hasCapability(Capability<?> capability, @Nullable EnumFacing faci
 
     @Override
     public int receiveEnergy(int maxReceive, boolean simulate) {
-        int result = storage.receiveEnergy(maxReceive, simulate);
-        energyBuffer = storage.getEnergyStored();
+        int energyReceived = Math.min((int)(energyMax - energyBuffer),maxReceive);
+        if (!simulate)
+            energyBuffer += energyReceived;
         markChanged();
-
-        return result;
+        return energyReceived;
     }
 
     @Override
     public int extractEnergy(int maxExtract, boolean simulate) {
-        int result = storage.extractEnergy(maxExtract, simulate);
-        energyBuffer = storage.getEnergyStored();
+        int energyExtracted = Math.min((int) energyBuffer, maxExtract);
+        if (!simulate)
+            energyBuffer -= energyExtracted;
         markChanged();
-
-        return result;
+        return energyExtracted;
     }
 
     @Override
     public int getEnergyStored() {
-        return storage.getEnergyStored();
+        if (energyBuffer >= Integer.MAX_VALUE) {
+            return Integer.MAX_VALUE;
+        }
+        return (int) energyBuffer;
     }
 
     @Override
     public int getMaxEnergyStored() {
-        return storage.getMaxEnergyStored();
+        if (energyMax >= Integer.MAX_VALUE) {
+            return Integer.MAX_VALUE;
+        }
+        return (int) energyMax;
     }
 
     @Override
     public boolean canExtract() {
-        return storage.canExtract();
+        return true;
```

Comment:
Should be false as this is not supposed to give out rf

Reply:
@Liach is correct on this.

Reply:
yea! I agree, I kept because storage.canExtract(); 
returns true. but it's better to change to false.